src/backend/distributed/planner/intermediate_result_pruning.c

```diff
@@ -101,11 +98,13 @@ RecordSubplanExecutionsOnNodes(HTAB *intermediateResultsHash,
 		IntermediateResultsHashEntry *entry = SearchIntermediateResult(
 			intermediateResultsHash, resultId);
 
-		/* no need to traverse the whole plan if all the workers are hit */
-		if (list_length(entry->nodeIdList) == workerNodeCount)
+		/*
+		 * There is no need to traverse the whole plan if the intermediate result
```

Comment:
I get this log:
```
DEBUG:  Subplan 61_1 will be sent to localhost:9700
DEBUG:  Subplan 61_1 will be sent to localhost:9701
DEBUG:  Subplan 61_1 will be sent to localhost:9702
DEBUG:  Subplan 61_1 will be written to local file
DEBUG:  Subplan 61_2 will be written to local file
```

I'm operation on `localhost:9700`, is this expected to see both written to local file and `will be sent to localhost:9700`?

Also, I already have 3 nodes only, why don't we see `Subplan %s is used in all workers`? 

I'm asking whether are these logging issues or we're actually sending the file twice?

the query that I use on MX worker is:

```SQL
WITH foo AS (SELECT * FROM users_table OFFSET 5) SELECT user_id, count(*) FROM foo JOIN users_table USING (user_id) GROUP BY user_id HAVING (max(foo.value_1) > (SELECT max(user_id) FROM foo)) ORDER BY 1 DESC LIMIT 5;
```

Reply:
I fixed this